lib/wallet.py

```diff
@@ -351,8 +352,8 @@ def save_transactions(self, write=False):
 
     def activate_slp(self):
         # This gets called in two situations:
-        # - Upon wallet startup once GUI is loaded, it checks config to see if SLP should be enabled.
-        # - During wallet operation, SLP can be freely enabled/disabled by user.
+        # - Upon wallet startup, it checks config to see if SLP should be enabled.
+        # - During wallet operation, on a network reconnect, to "wake up" the validator -- According to JSCramer this is required.  TODO: Investigate why that is
```

Comment:
@cculianu a user was reporting that the validator would not start back up after a network loss/reconnection.  Calling this method upon network reconnection gets the validator going again.  

You can reproduce by commenting out the `activate_slp()` call at `_slp_callback_on_status()`.  Then disable the network, let the validator keep trying to validate until it eventually quits trying.  Then reconnect the network and the validator does not start itself back up.

Reply:
Ah, cool. 

I believe you -- I was just wondering if perhaps the validator can just "manage itself" in that situation and not need hand-holding; if it makes sense and isn't too much code to do that -- I'll look into it.  

It is not a high priority in either case, however it would make using the validator require less "expert knowledge" on its idiosyncrasies is all.

Reply:
Right, do you know why the validator is giving up in the first place?  That's something I didn't really look into, i just assume its related to a timeout or something but not sure.

Reply:
No idea.. will look into it. Hence that comment, ha ha. :)

Reply:
When/if I find out, I'll be sure to tell you though...